update/common/scripts/5.4/cleanupfieldvaluerelations.php

```diff
@@ -0,0 +1,249 @@
+            "For more details see https://jira.ez.no/browse/EZP-22408.",
+        'use-session' => false,
+        'use-modules' => false,
+        'use-extensions' => true
+    )
+);
+$script->startup();
+
+$options = $script->getOptions(
+    "[n][v][iteration-sleep:][iteration-limit:]",
+    "",
+    array(
+        'iteration-sleep' => 'Sleep duration between batches, in seconds (default: 1)',
+        'iteration-limit' => 'Batch size (default: 100)',
+        'n' => 'Do not wait 30 seconds before starting',
+    )
+);
+$optIterationSleep = (int)$options['iteration-sleep'] ?: 1;
+$optIterationLimit = (int)$options['iteration-limit'] ?: 100;
+$verboseLevel = $script->verboseOutputLevel();
+
+$script->initialize();
+
+$db = eZDB::instance();
+
+if ( !$db->isConnected() )
+{
+    $cli->error( "Can't initialize database connection.\n" );
+    $script->shutdown( 1 );
+}
+
+$relationCount = $result = $db->arrayQuery(
+    "SELECT ( COUNT( * ) ) AS relation_count
+     FROM ezcontentobject_attribute
+     LEFT JOIN ezcontentobject ON ezcontentobject_attribute.data_int = ezcontentobject.id
+     WHERE ezcontentobject_attribute.data_type_string = 'ezobjectrelation' AND
+           ezcontentobject_attribute.data_int IS NOT NULL AND
+           ezcontentobject.id IS NULL"
+);
+$relationCount = $relationCount[0]["relation_count"];
+
+$relationListCount = $result = $db->arrayQuery(
+    "SELECT ( COUNT( * ) ) AS relationlist_count
+     FROM ezcontentobject_attribute
+     WHERE ezcontentobject_attribute.data_type_string = 'ezobjectrelationlist'"
+);
+$relationListCount = $relationListCount[0]["relationlist_count"];
+
+$cli->warning( "This script removes relation data towards non-existing Content from field values." );
+$cli->warning( "Only Relation and RelationList fieldtypes field values will be processed." );
+$cli->warning( "For more details about this cleanup, see https://jira.ez.no/browse/EZP-22408." );
+$cli->output();
+$cli->output( "Found total of {$relationCount} Relation and {$relationListCount} RelationList field values to process." );
+
+if ( $relationCount == 0 && $relationListCount == 0 )
+{
+    $cli->output( "Nothing to process, exiting." );
+    $script->shutdown( 0 );
+}
+
+$cli->output();
+$cli->output( "Relation field values will be updated in a single update" );
+$cli->output( "RelationList field values will be processed in batches of {$optIterationLimit}, with {$optIterationSleep} seconds pause between the batches." );
+$cli->warning( "Depending on the number of RelationList field values found, this script can potentially run for a very long time." );
+
+if ( !$options['n'] )
+{
+    $cli->output();
+    $cli->warning( "You have 30 seconds to break the script before actual processing starts (press Ctrl-C)." );
+    $cli->warning( "Execute the script with '-n' switch to skip this delay." );
+    sleep( 30 );
+}
+
+function processRelationRows()
+{
+    global $db, $cli, $relationCount;
```

Comment:
Fixed directly in https://github.com/ezsystems/ezpublish-legacy/commit/f5e28c0394203b410fdba583bb752717fc5da891.